new/resource_metadata.md

```diff
@@ -0,0 +1,122 @@
+is unsafe. Therefore, a cookbook's metadata should contain the details of
+the resources contained in the cookbook.
+
+## Motivation
+
+    As a user of Supermarket,
+    I want to view details of resources,
+    so that I can write great cookbooks.
+
+    As a cookbook author,
+    I want code completion of any resource,
+    so that I can write great cookbooks.
+
+## Specification
+
+A new `resources` field will be added to the metadata json format.
+Resources are a set of resource descriptions generated by the
+`chef-resource-inspector`. Each resource has the following keys:
+
+* Description: The description of the resource
+* Actions: The list of actions available on the resource
+* Examples: Optional code examples
+* Introduced: The cookbook version that the resource was introduced in.
+* Properties: A list of properties that the resource supports.
+
+Each property has the following keys:
+* Name: Name of the property.
+* Is: The types that the property accepts.
+* Description: A description of the property.
+* Introduced: The cookbook version that added the property
+* Deprecated: Is the property deprecated?
+
+Example:
+```json
+{
+  "resources":
+  {
+    "rhsm_errata": {
+      "description": "Use the rhsm_errata resource to install packages associated with a given Red Hat Subscription Manager Errata ID. This is helpful if packages to mitigate a single vulnerability must be installed on your hosts.",
+      "actions": [
+        "nothing",
+      "install"
+      ],
+      "examples": null,
+      "introduced": "14.0",
+      "properties": [
+      {
+        "name": "name",
+        "description": null,
+        "introduced": null,
+        "is": "String",
+        "deprecated": false
+      },
+      {
+        "name": "retries",
+        "description": null,
+        "introduced": null,
+        "is": "Integer",
+        "deprecated": false
+      },
+      {
+        "name": "retry_delay",
+        "description": null,
+        "introduced": null,
+        "is": "Integer",
+        "deprecated": false
+      },
+      {
+        "name": "sensitive",
+        "description": null,
+        "introduced": null,
+        "is": [
+          "TrueClass",
+        "FalseClass"
+        ],
+        "deprecated": false
+      },
+      {
+        "name": "ignore_failure",
+        "description": null,
+        "introduced": null,
+        "is": [
+          "TrueClass",
+        "FalseClass"
+        ],
+        "deprecated": false
+      },
+      {
+        "name": "errata_id",
+        "description": "An optional property for specifying the errata ID if not using the resource's name.",
+        "introduced": null,
+        "is": "String",
+        "deprecated": false
+      }
+      ]
+    }
+  }
+}
+```
+## Downstream Impact
+
+* Stove will be updated to run `chef-resource-inspector` as part of the upload process.
+* Supermarket will add a new tab displaying resources in a cookbook.
```

Comment:
Should there be a mention of `knife cookbook site/supermarket` having this behavior added, too?